src/Components/ContextualMenu/ContextualMenu.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { ReactElement } from "react"
+
+import MoreIcon from "@artsy/icons/MoreIcon"
+import { Clickable, Dropdown } from "@artsy/palette"
+import { ContextualMenuItem } from "Components/ContextualMenu/ContextualMenuItem"
+
+interface ContextualMenuProps {
+  /** Supply `ContextualMenuItem`s and `ContextualMenuDivider`s as needed  */
+  children: React.ReactNode
+}
+
+/**
+ * Creates a contextual menu by composing a MoreIcon (⋯) trigger
+ * and a Dropdown component from Palette.
+ */
+export const ContextualMenu: React.FC<ContextualMenuProps> = ({ children }) => {
+  return (
+    <Dropdown
+      openDropdownByClick
+      placement="bottom-end"
+      offset={0}
+      dropdown={({ onHide }) => {
+        /*
+         * Each ContextualMenuItem must be able to close this
+         * parent ContextualMenu's Dropdown when it is clicked.
+         */
+        return React.Children.map(children, (child: ReactElement) => {
+          if (child.type === ContextualMenuItem) {
+            const onClick = () => {
+              onHide()
+              child.props.onClick?.()
+            }
+            return React.cloneElement(child, {
+              onClick,
+            })
+          } else {
+            return child
+          }
+        })
```

Comment:
**question**: Maybe it would be better if we **only** allow passing `ContextualMenuItem` as children? Otherwise, if we pass another component, we will not be able to close the dropdown menu (see attached video)

### Demo
https://user-images.githubusercontent.com/3513494/220097110-75582839-bf27-4aa9-80ea-ad2df878c9a9.mp4

### Code
```javascript
<>
  {isDeleteModalOpen && (
    <DeleteSavesModal
      collection={collection}
      setIsDeleteModalOpen={setIsDeleteModalOpen}
    />
  )}

  <ContextualMenu>
    <Clickable bg="red">Hello</Clickable>

    <ContextualMenuItem onClick={handleEditList}>
      {t("collectorSaves.contextualMenu.edit")}
    </ContextualMenuItem>

    <ContextualMenuDivider />

    <ContextualMenuItem onClick={handleDeleteList}>
      {t("collectorSaves.contextualMenu.delete")}
    </ContextualMenuItem>
  </ContextualMenu>
</>
```